src/server/.tests/ws-routes/controllers/Friends.spec.js

```diff
@@ -0,0 +1,482 @@
+			suite.friendsInstance.addFriends = sinon.stub().resolves()
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+				socket.on(CLIENT_EVENTS.INVITATION_CONFIRMATION, data => {
+					return suite.friendsInstance.invitaitonConfirmation(data, socket, suite.connectionsMock)
+						.then(asserations)
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.INVITATION_CONFIRMATION, suite.REQUEST_MOCK)
+
+			//then
+			function asserations() {
+				const expectedInvitatingUsername = suite.DUMMY_INVITATING_USERNAME
+				const expectedInvitatedUsername = suite.DUMMY_USERNAME
+				const expectedNotificationType = CLIENT_EVENTS.INVITATION_CONFIRMATION
+				sinon.assert.calledWith(suite.utilsMock.addNotification, suite.notificationModelMock, suite.userModelMock, expectedInvitatedUsername, expectedInvitatingUsername, expectedNotificationType)
+				done()
+			}
+		})
+	})
+
+	describe('#isNotFriendsAlready', () => {
+		beforeEach(() => {
+			suite.userModelMock = {
+				findOne: sinon.stub()
+			}
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: suite.userModelMock
+			})
+		})
+
+		it('should reject promise when invitating username is on invitated username friends list', done => {
+			//given
+			const INVITATING_USERNAME = suite.DUMMY_USERNAME
+			const INVITATED_USERNAME = 'INVITATED_USERNAME'
+
+			const QUERY_FEEDBACK_MOCK = {
+				friends: [
+					{
+						username: INVITATING_USERNAME
+					}
+				]
+			}
+
+			suite.queryResult = {
+				exec: sinon.stub().resolves(QUERY_FEEDBACK_MOCK)
+			}
+
+			suite.userModelMock.findOne.returns(suite.queryResult)
+
+			//when
+			suite.friendsInstance.isNotFriendsAlready(INVITATING_USERNAME, INVITATED_USERNAME)
+				//then
+				.catch(done)
+
+		})
+	})
+
+	describe('#addFriends', () => {
+		it('should push firend username to firends array', done => {
+			//given
+			const DUMMY_USERNAME_2 = 'DUMMY_USERNAME_2'
+			const COLLECTION_MOCK = [
+				{
+					save: sinon.stub(),
+					username: suite.DUMMY_USERNAME,
+					friends: []
+				},
+				{
+					save: sinon.stub(),
+					username: DUMMY_USERNAME_2,
+					friends: []
+				},
+			]
+			suite.usersCollectionMock = _.cloneDeep(COLLECTION_MOCK)
+			suite.userModelMock = {
+				find: sinon.stub()
+			}
+			const queryResultMock = {
+				exec: sinon.stub().resolves(suite.usersCollectionMock)
+			}
+			suite.userModelMock.find.returns(queryResultMock)
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: suite.userModelMock
+			})
+
+			//when
+			const invitatingUsername = suite.DUMMY_USERNAME
+			const invitatedUSername = DUMMY_USERNAME_2
+			suite.friendsInstance.addFriends(invitatingUsername, invitatedUSername)
+				.then(() => then())
+
+			//then
+			function then() {
+				let expectedUserCollectionMock = _.cloneDeep(COLLECTION_MOCK)
+				expectedUserCollectionMock[0].friends.push({username: DUMMY_USERNAME_2})
```

Comment:
Can you just represent it in declarative manner?

Reply:
ok